react/features/app/functions.web.js

```diff
@@ -99,23 +131,24 @@ export function _getRouteToRender(stateOrGetState: Object | Function) {
  * {@link getState} function.
  * @param {ReactElement} component - Current route component to render.
  * @private
- * @returns {ReactElement} If any of the pre-defined rules is satisfied, returns
- * intercepted component.
+ * @returns {Promise<ReactElement>} If any of the pre-defined rules is
+ * satisfied, returns intercepted component.
  */
 function _interceptComponent(
         stateOrGetState: Object | Function,
         component: React$Element<*>) {
-    let result;
     const state = toState(stateOrGetState);
 
-    for (const rule of _INTERCEPT_COMPONENT_RULES) {
-        result = rule(state);
-        if (result) {
-            break;
-        }
-    }
+    const promises = [];
+
+    _INTERCEPT_COMPONENT_RULES.forEach(rule => {
+        promises.push(rule(state));
+    });
 
-    return result || component;
+    return Promise.all(promises).then(
+        results =>
+            results.find(result => typeof result !== 'undefined') || component,
+        () => Promise.resolve(component));
```

Comment:
It's good the error case is being handled. What's an example case where it might be exercised?

Reply:
Currently there isn't. I added it just in case. Do you want me to remove it?

Reply:
No don't remove it. It's good to handle errors. I was curious what an example case could be.